.buildkite/scripts/post-release.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+TAG="v{$1}"
+
+set_git_config() {
+    git config user.name "${GITHUB_USERNAME_SECRET}"
+    git config user.email "${GITHUB_EMAIL_SECRET}"
+}
+
+tag_commit() {
+  echo "Tagging commit ${BUILDKITE_COMMIT}"
+  git tag -a -m "${BUILDKITE_COMMIT}" "${TAG}"
+}
+
+git_push_with_auth() {
+  echo "Pushing tag ${TAG}"
+  retry 3 git push https://${GITHUB_USERNAME_SECRET}:${GITHUB_TOKEN_SECRET}@github.com/elastic/golang-crossbuild.git ${TAG}
+}
+
+if [ ${TAG_EXISTS} ]; then
```

Comment:
Looking at the declaration of `TAG_EXISTS` [earlier in the code](https://github.com/elastic/golang-crossbuild/pull/326/files#diff-68c0021e761ebe6b8a42d32dfdf3dda2200f0f2fee50106c85c9aea530ccf3c3R132-R135), it seems to either return `"true"` or `"false"` (as in string, not boolean).

```
$ foo="true"
$ if [ $foo ]; then echo "is true"; else echo "is false"; fi
is true

$ foo="false"
$ if [ $foo ]; then echo "is true"; else echo "is false"; fi
is true
```

We should either change it to `if $TAG_EXISTS; then` or (more explicit and preferable IMO)  `if [[ $TAG_EXISTS == "true" ]]; then`

(nit: let's also be consistent with the use of double square brackets)